### PR TITLE
Update: curly multi-or-nest flagging semis on next line (fixes #12370)

### LINKED ISSUE #12370: curly: multi-or-nest option reports an error for a single-statment block with a semicolon on the next line
**Tell us about your environment**
* **ESLint Version:** 6.5.1
* **Node Version:** 12.11.0
* **npm Version:** 6.11.3

**What parser (default, Babel-ESLint, etc.) are you using?** default

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```json
{
  "rules": {
    "curly": [2, "multi-or-nest"]
  }
}
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

I used the following code in `file.js`:
```js
if (true) console.log('some statement')
;(function () {})()
```

The issue occurs whenever the statement in the if block (or any block) has the semicolon on the following line.

Then I linted it:
```bash
eslint file.js
```

**What did you expect to happen?**
No error would occur as there is only 1 statement in the if block.

**What actually happened? Please include the actual, raw output from ESLint.**
I got the following error:
```
file.js
  1:1  error  Expected { after 'if' condition  curly

✖ 1 problem (1 error, 0 warnings)
  1 error and 0 warnings potentially fixable with the `--fix` option.
```

When fixing it with `eslint --fix file.js`, it gets changed to
```js
if (true) {console.log('some statement')
;}(function () {})()
```
which conflicts with the `"semi": [2, "never"]` rule as that would change it to:
```js
if (true) {console.log('some statement')
}(function () {})()
```
which would cause `curly` to report `Unnecessary { after 'if' condition`.

**Are you willing to submit a pull request to fix this bug?**
Yes.

### PR DESCRIPTION
**Pull request by cherryblossom000**
**What is the purpose of this pull request? (put an "X" next to item)**

[x] Bug fix

Fixes #12370.

**What changes did you make? (Give an overview)**

I made the `isOneLiner` function use the last token excluding the semicolon instead of just the last token, like the `isCollapsedOneLiner` function, which stopped multi-or-nest flagging code like the following as one line.
```js
console.log('foo')
;
```

**Is there anything you'd like reviewers to focus on?**

No.

### CODE REVIEW

**Comment by mdjermanovic** [comment]
Thanks for the PR! Marking as accepted, as the issue is accepted.

Can this change also produce more warnings? For example, in some rare situations like this:

```js
/* eslint curly: ["error", "multi-or-nest"] */

if (foo) {
  bar()
;
}
```

**Comment by cherryblossom000** [comment]
Yes, this change does produce those warnings. I added more tests for that.

File: lib/rules/curly.js [hunk 1]
```diff
@@ -97,10 +97,11 @@ module.exports = {
...
          */
         function isOneLiner(node) {
-            const first = sourceCode.getFirstToken(node),
-                last = sourceCode.getLastToken(node);
+            const first = sourceCode.getFirstToken(node);
+            const last = sourceCode.getLastToken(node);
+            const lastExcludingSemicolon = astUtils.isSemicolonToken(last) ? sourceCode.getTokenBefore(last) : last;
 
-            return first.loc.start.line === last.loc.end.line;
+            return first.loc.start.line === lastExcludingSemicolon.loc.end.line;
```
Comment (mdjermanovic) [inline lib/rules/curly.js]: What will happen with an empty statement (only the `;`).  For example, would this code have a warning:

```js
/* eslint curly: ["error", "multi-or-nest"] */

if (foo) 
;
```

Comment (cherryblossom000) [reply lib/rules/curly.js]: Yes, that code would have produced a warning. I just added
```js
if (node.type === "EmptyStatement") {
    return true;
}
```
at the beginning of the function to fix it.

Comment (mdjermanovic) [reply lib/rules/curly.js]: Thanks!

**Comment by mdjermanovic** [comment]
One small detail I missed before, sorry - 'curly' is the rule's name and should start with a lowercase in commit message and PR title.

Other than that, everything else LGTM!

**Comment by mdjermanovic** [comment]
> One small detail I missed before, sorry - 'curly' is the rule's name and should start with a lowercase in commit message and PR title.
> 
> Other than that, everything else LGTM!

Actually, please disregard the 'commit message' part. 

It's okay to change just the PR title because this PR has more than 1 commit, so the PR title will be used as the merge commit message.

**Comment by cherryblossom000** [comment]
@kaicataldo Sorry, ignore the review request. That was accidental.

**Comment by kaicataldo** [comment]
Thanks for contributing to ESLint!

**Pull request merged**